stdlib/public/core/Unicode.swift

```diff
@@ -276,6 +262,9 @@ extension Unicode.UTF8: UnicodeCodec {
     if _fastPath(s == 0) { return }
     processCodeUnit(UInt8(truncatingIfNeeded: s) &- 0x01)
   }
+}
+
+extension Unicode.UTF8 {
```

Comment:
The _non-deprecated_ APIs could be moved into UTF8.swift:
* `Unicode.UTF8.isContinuation(_:)`
* `Unicode.UTF8._nullCodeUnitOffset(in:)`
* `Unicode.UTF8._decodeOne(_:)` (used in [UnicodeLongTest.swift][])

The _deprecated_ APIs could be moved into MigrationSupport.swift:
* `UnicodeDecodingResult` enum,
* `UnicodeCodec` protocol and conformances.

The _unused_ APIs could be moved into LegacyABI.swift:
* `Unicode.UTF16._decodeOne(_:)`
* `transcode(_:_:_:_:stopOnError:)` ([unavailable][] since Swift 3).

[unavailable]: <https://github.com/apple/swift/blob/swift-3.0-branch/stdlib/public/core/Unicode.swift#L1152-L1164>
[UnicodeLongTest.swift]: <https://github.com/apple/swift/blob/master/validation-test/stdlib/UnicodeLongTest.swift>